apps/service_providers/llm_service/runnables.py

```diff
@@ -330,12 +331,21 @@ def invoke(
                 input_dict["thread_id"] = current_thread_id
             input_dict["instructions"] = self.adapter.get_assistant_instructions()
             thread_id, run_id = self._get_response_with_retries(merged_config, input_dict, current_thread_id)
+            dispatch_custom_event(
+                "OpenAI Assistant Info",
+                {
+                    "name": self.adapter.assistant.name,
+                    "id": self.adapter.assistant.assistant_id,
+                    "thread_id": thread_id,
+                    "run_id": run_id,
+                },
+            )
             ai_message, annotation_file_ids = self._get_output_with_annotations(thread_id, run_id)
             ai_message_metadata = self.adapter.get_output_message_metadata(annotation_file_ids)
 
             if not current_thread_id:
-                self.adapter.thread_id = thread_id
-
+                self.adapter.update_thread_id(thread_id)
```

Comment:
Today I learned that one can return inside the try and the `finally` block still gets executed.